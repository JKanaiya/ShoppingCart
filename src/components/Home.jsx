import { useState } from "react";
import Shop from "./Shop";
import Cart from "./Cart";
import fetcherWithFetch from "../fetcherWithFetch";

const Home = function ({ initialItems = [] }) {
  const [cartItems, setCartItems] = useState(initialItems);

  const addItemToCart = function (item) {
    setCartItems([...cartItems, item]);
  };

  const editItemCount = function (item) {
    const b = cartItems;
    b.find((n) => n.id == item.id).count = item.count;
    setCartItems(b);
  };

  const deleteItem = function (item) {
    setCartItems(
      cartItems.toSpliced(cartItems.findIndex((n) => n.id == item.id)),
      1,
    );
  };

  return (
    <div>
      <h2>aaa</h2>
      <Shop addItemToCart={addItemToCart} fetcherWithFetch={fetcherWithFetch} />
      <Cart
        cartItems={cartItems}
        editItemCount={editItemCount}
        deleteItem={deleteItem}
      />
    </div>
  );
};

export default Home;
