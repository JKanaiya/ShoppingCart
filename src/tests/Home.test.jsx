import { describe, it, vi, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Home from "../components/Home";

vi.mock(
  "../components/Cart.jsx",
  () =>
    ({ cartItems, editItemCount, deleteItem }) => (
      <div key={cartItems}>
        {cartItems.map((item) => (
          <div>
            <button onClick={() => editItemCount(item)}>Add an Item</button>
            <button onClick={() => editItemCount(item)}>
              Reduce Item Count
            </button>
            <button onClick={() => deleteItem(item)}>Delete Item</button>
          </div>
        ))}
      </div>
    ),
);

describe("Home Component", () => {
  it("Renders the Home Page", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
