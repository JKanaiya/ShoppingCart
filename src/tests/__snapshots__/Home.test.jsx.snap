// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > Renders the Home Page 1`] = `
<div>
  <div>
    <h2>
      aaa
    </h2>
    <div>
      <div>
        Loading...
      </div>
    </div>
    <div>
      Cart
    </div>
  </div>
</div>
`;
